nservicebus/messaging/evolving-contracts.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Evolving messages contracts
+reviewed: 2016-11-08
+component: Core
+---
+
+In message-based systems the messages are part of the contract, which defines how services communicate with each other.
+
+Evolving contracts over time is not an easy task and the appropriate strategy should be defined individually for each system. Generally that problem can't be resolved on the infrastructure level, therefore NServiceBus users need to analyze their systems, consider how they expect them to evolve and define the strategy which will make the most sense in their particular circumstances.
+
+This article presents basic guidelines for choosing contracts evolution strategy, avoiding common mistakes and ensuring the contracts will be easy to evolve over time.
+
+
+## Designing contracts
+
+
+### Messages
+
+Messages should ideally be:
+
+ * [**Plain Old CLR Objects** (POCO)](https://en.wikipedia.org/wiki/Plain_Old_CLR_Object).
+ * **Small and focused**. Messages should carry the minimum amount of information and should be split if they get too big.
+ * **Serve a single purpose**. Messages shouldn't be used for anything else than carrying data. For example, messages shouldn't be used as data access objects or as UI binding models.
+
+
+### Assemblies
```

Comment:
I've seen projects that just share the C# files should we mention that as well? (they used source only packages)

The upside of that is that you won't need to worry about the NServiceBus version the assembly is compiled against and the mess of assembly redirect that ensues